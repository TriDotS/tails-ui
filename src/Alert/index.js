import styled, { keyframes } from 'styled-components'

import React from 'react'

const flip = keyframes`
  0% {
    transform: perspective(400px) rotateX(90deg);
    opacity: 0;
  }
  40% {
    transform: perspective(400px) rotateX(-10deg);
  }
  70% {
    transform: perspective(400px) rotateX(10deg);
  }
  100% {
    transform: perspective(400px) rotateX(0deg);
    opacity: 1;
  }
`

const Alert = styled.div`
  background: ${props => props.backgroundColor || 'red'};
  color: ${props => props.fontColor || 'white'};
  font-weight: ${props => (props.boldText ? 'bold' : 'normal')};
  padding: 20px;
  border-radius: ${props => (props.rounded ? '3px' : '0px')};
  animation: ${flip} 1.2s;
`

export default ({ children, backgroundColor, fontColor, rounded, boldText }) =>
  <Alert
    backgroundColor={backgroundColor}
    fontColor={fontColor}
    rounded={rounded}
  >
    {children}
  </Alert>
