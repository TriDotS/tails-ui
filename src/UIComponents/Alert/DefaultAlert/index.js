import DefaultAlert from './DefaultAlert'

export default DefaultAlert
