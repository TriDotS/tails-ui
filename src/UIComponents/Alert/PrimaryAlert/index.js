import PrimaryAlert from './PrimaryAlert'

export default PrimaryAlert
