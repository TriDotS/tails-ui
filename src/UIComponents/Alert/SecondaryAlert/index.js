import SecondaryAlert from './SecondaryAlert'

export default SecondaryAlert
