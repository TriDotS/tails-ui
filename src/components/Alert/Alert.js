import styled, { keyframes } from 'styled-components'

import React from 'react'

const flip = keyframes`
  0% {
    transform: rotateX(90deg);
    opacity: 0;
  }
  100% {
    transform: rotateX(0deg);
    opacity: 1;
  }
`

const Alert = styled.div`
  background: ${props => props.backgroundColor || 'red'};
  color: ${props => props.fontColor || 'white'};
  font-weight: ${props => (props.boldText ? 'bold' : 'normal')};
  padding: 20px;
  border-radius: ${props => (props.rounded ? '3px' : '0px')};
  box-shadow: ${props =>
    props.shadowed
      ? 'rgba(0, 0, 0, 0.16) 0px 3px 10px, rgba(0, 0, 0, 0.23) 0px 3px 10px'
      : 'none'};
  animation: ${flip} 0.5s;
`

export default ({
  children,
  backgroundColor,
  fontColor,
  rounded,
  boldText,
  shadowed,
}) =>
  <Alert
    backgroundColor={backgroundColor}
    fontColor={fontColor}
    rounded={rounded}
    boldText={boldText}
    shadowed={shadowed}
  >
    {children}
  </Alert>
