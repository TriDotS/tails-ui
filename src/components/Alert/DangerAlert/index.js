import DangerAlert from './DangerAlert'

export default DangerAlert
