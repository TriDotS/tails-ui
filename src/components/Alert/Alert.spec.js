import Alert from './Alert'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Alert />, div)
})

it('generate alert snapshot', () => {
  const component = renderer.create(<Alert />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
