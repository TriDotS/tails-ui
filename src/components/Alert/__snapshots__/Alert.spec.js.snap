// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate alert snapshot 1`] = `
<div
  className="sc-bdVaJa lhTmjz"
/>
`;
